qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDiagnostics.java

```diff
@@ -1134,36 +1135,69 @@ private ResolvedJavaTypeInfo validateIterable(Part part, Section ownerSection,
 	 * @param projectUri  the project Uri.
 	 * @param diagnostics the diagnostic list.
 	 */
-	private void validateCaseSectionParameters(Section caseSection, Section whenSection, ResolvedJavaTypeInfo whenType,
-			String projectUri, List<Diagnostic> diagnostics) {
+	private void validateCaseSectionParameters(CaseSection caseSection, Section whenSection,
+			ResolvedJavaTypeInfo whenType, String projectUri, List<Diagnostic> diagnostics) {
 		String expression = whenSection.getExpressionContent();
 		List<Parameter> parameters = caseSection.getParameters();
 		String whenTypeName = whenType.getSignature();
+		Parameter firstParameter = parameters.get(0);

```

Comment:
If you have just `{#case}` I think you will have an expcetion.

I suggest that you try if parameters.isEmpty() at first and in this use you report a diagnostic with a new error code `CaseParameterRequired` and you do a return in this usecase.